includes/admin/class-amp-editor-blocks.php

```diff
@@ -105,6 +105,10 @@ public function enqueue_block_editor_assets() {
 			true
 		);
 
-		wp_add_inline_script( 'amp-editor-blocks', sprintf( 'ampEditorBlocks.boot();' ) );
+		wp_add_inline_script(
+			'amp-editor-blocks',
+			'wp.i18n.setLocaleData( ' . wp_json_encode( gutenberg_get_jed_locale_data( 'amp' ) ) . ', "amp" );' . sprintf( 'ampEditorBlocks.boot();' ),
```

Comment:
At this moment it's just including the `amp.pot` file created by `babel`, not including the PHP file translations, wondering if we should create a separate locale for the blocks and load only that specific file within the blocks instead?

Reply:
That's one of the questions raised in https://make.wordpress.org/core/2018/05/01/javascript-internationalization-the-missing-pieces/ under “Loading only specific set of translations”.

> Use a different text domain per JavaScript module. Export a JSON file per text domain. This is appealing, but has to be ruled out quickly: the text domain is not known to GlotPress and is not stored in the database or anything.

If Babel is just creating an `amp.pot` for the JS strings, then we can just use that as-is and ignore the `amp.pot` for all other purposes. In other words, maybe there could be an alternative to `gutenberg_get_jed_locale_data()` which allows you to pass a path to a POT file instead (or maybe there is an underlying function that could be used instead). And then this POT file would _not_ be used for translating the plugin as a whole. Instead, Babel should generate a dummy PHP file that contains all of the strings in it for translation by GlotPress. You can see that Gutenberg does this by generating a `languages/gutenberg-translations.php` file in addition to the `languages/gutenberg.pot`.

Not sure if this is possible, but it's what I'd be looking into.

Reply:
If I understand correctly then the following is true on using different text domain per JS module, however, it should be possible for separating the editor blocks translation file for all the blocks from the rest in case of this plugin since `babel` is creating the `.pot` file based on the `block` directory files only.
>Use a different text domain per JavaScript module. Export a JSON file per text domain. This is appealing, but has to be ruled out quickly: the text domain is not known to GlotPress and is not stored in the database or anything.

Added a commit where `amp-js` locale is registered for editor blocks JS translation and is creating a separate `amp-js.pot` file for translation of the blocks, leaving `amp.pot` available for PHP and any other translations outside of `blocks` directory: 8ac7759. At least it seems to be working locally when translating some block-related strings within `amp-js-et.po` and strings from PHP files within `amp-et.po`.

Do you think this could be suitable setup for the plugin?

Reply:
I don't know how well this will work with GlotPress on translate.wordpress.org. My understanding is that the POT files are not read and that only the strings used in PHP files are parsed and made available for translation. How would GlotPress be told of the POT files in the first place? I could very well be wrong. @swissspidy Do you have insights to share on the approach?

Reply:
Regarding telling about POT files: it‘s one of the possible features for JS 18N, see our latest chat summary.

Reply:
For reference: https://github.com/swissspidy/gutenberg-i18n-block

Reply:
Good to know that the JS i18n defining is happening right now, taking a look at the chat summary. Thanks for the clarification about the limitations of the current system.